### ISSUE TITLE
Incomplete points drawing at different zoom levels

### ISSUE THREAD

**Issue by ivanprado** [splashblot/dronedb#20]
Look what happen [here](https://alpha.tileo.co/user/ivan/builder/d66cec21-c13e-42df-8af9-4945dd21486b/embed) when zooming out on a dataset with many points. 
![image](https://cloud.githubusercontent.com/assets/895720/24611069/28617862-1881-11e7-9b67-932999462f43.png)
![image](https://cloud.githubusercontent.com/assets/895720/24611087/36b9fd94-1881-11e7-8f82-1f542388f1e4.png)
![image](https://cloud.githubusercontent.com/assets/895720/24611078/2e997f86-1881-11e7-9cc0-323d8b8ae4bb.png)

For some reason, at smaller zoom levels drawing is not completed. is there a limit on the number of shapes carto can draw?

**Comment by apercas** [comment]
**UPDATE**:

I checked wether this happens or it doesn't on CARTO (either old editor or the builder) with the own dataset used here and I confirmed that it works perfectly.
After this we discussed that maybe it is a matter of how many points we're displaying and applied a `LIMIT` clause to the `SELECT` and then with a `WHERE cartodb_id > x` (x: 50000, 75000, 100000...)  query on the builder to check if the points displayed changed if I asked for different points. It did so.

Once the difference on the amount of points was clear I asked on CARTO's Google Group https://groups.google.com/forum/#!topic/cartodb/nzrti8xF73M where they informed me to check a possible DB timeout (I had already checked that).

Thinking it is related with the painting we started checking how Windshaft / Mapnik handles the data for the tiles it serves.

I was currently debugging L250 of `/dronedb/node_modules/cartodb.js/src/vis/vis.js`, the `_onMapInstanceCreated` method. No conclusion yet whatsoever 🙁
